blockchain-source/src/cardano/event.rs

```diff
@@ -0,0 +1,129 @@
+use crate::EventObject;
+use anyhow::Context;
+use cardano_sdk::protocol::SerializedBlock;
+use dcspark_core::{BlockId, BlockNumber, SlotNumber};
+use serde::de::Visitor;
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub enum CardanoNetworkEvent<Block, Tip> {
```

Comment:
probably makes sense to merge with CardanoEventType in carp

Reply:
This is a bit more general, but it makes sense. I could add a variant for rollback events, and leave it unused in the source for now.

The multiverse source could then take a `CardanoNetworkEvent<Block, Tip, Rollback>` and transform it into `CardanoNetworkEvent<Block, Tip, ()>` in the bridge.

Reply:
this is a point to discuss by the way. If we decide to add one more network event will we introduce 4th template parameter? I think we can generalize just the block, not every type of event. The block can be parsed by different libraries and have any type after that, so to keep the event structure (tips, rollbacks) it needs to be generic. However, I don't see a point where other events can change their type during propagation

Reply:
Well, it's not necessary to parametrize rollbacks, that suggestion was mostly so that you can know when you can just ignore it. But it can be handled in different ways if you need that (like creating a new type with less variants), so I think it should be ok to use a concrete type.

The `Tip` currently needs to be parametric though, because we need to compute the timestamp, and currently the `DeserializerSource` is the only one that knows how to do that. We can find a different solution for that, but I would prefer to do it after we finish the port, to make it easier to analyze.